stock_barcodes/wizard/stock_barcodes_read_picking.py

```diff
@@ -0,0 +1,343 @@
+        vals = super()._prepare_scan_log_values(log_detail=log_detail)
+        vals['picking_id'] = self.picking_id.id
+        if log_detail:
+            vals['log_line_ids'] = [(0, 0, {
+                'move_line_id': x[0],
+                'product_qty': x[1],
+            }) for x in log_detail.items()]
+        return vals
+
+    def remove_scanning_log(self, scanning_log):
+        for log in scanning_log:
+            for log_scan_line in log.log_line_ids:
+                if log_scan_line.move_line_id.state not in ['assigned',
+                                                            'confirmed']:
+                    raise ValidationError(_(
+                        'You can not remove an entry linked to a stock move '
+                        'line in state assigned or confirmed')
+                    )
+                qty = (log_scan_line.move_line_id.qty_done -
+                       log_scan_line.product_qty)
+                log_scan_line.move_line_id.qty_done = max(qty, 0.0)
+            self.picking_product_qty = sum(log.log_line_ids.mapped(
+                'move_line_id.move_id.quantity_done'))
+            log.unlink()
+
+    def action_undo_last_scan(self):
+        res = super().action_undo_last_scan()
+        log_scan = first(self.scan_log_ids.filtered(
+            lambda x: x.create_uid == self.env.user))
+        self.remove_scanning_log(log_scan)
+        return res
+
+
+class WizCandidatePicking(models.TransientModel):
+    """
+    TODO: explain
+    """
+    _name = 'wiz.candidate.picking'
+    _description = 'Candidate pickings for barcode interface'
+    # To prevent remove the record wizard until 2 days old
+    _transient_max_hours = 48
+
+    wiz_barcode_id = fields.Many2one(
+        comodel_name='wiz.stock.barcodes.read.picking',
+        readonly=True,
+    )
+    picking_id = fields.Many2one(
+        comodel_name='stock.picking',
+        string='Picking',
+        readonly=True,
+    )
+    wiz_picking_id = fields.Many2one(
+        comodel_name='stock.picking',
+        related='wiz_barcode_id.picking_id',
+        string='Picking',
+        readonly=True,
+    )
+    name = fields.Char(
+        related='picking_id.name',
+        readonly=True,
+        string='Picking',
+    )
+    partner_id = fields.Many2one(
+        comodel_name='res.partner',
+        related='picking_id.partner_id',
+        readonly=True,
+        string='Partner',
+    )
+    state = fields.Selection(
+        related='picking_id.state',
+        readonly=True,
+    )
+    date = fields.Datetime(
+        related='picking_id.date',
+        readonly=True,
+        string='Creation Date',
+    )
+    product_qty_reserved = fields.Float(
+        'Reserved', compute='_compute_picking_quantity',
+        digits=dp.get_precision('Product Unit of Measure'),
+        readonly=True,
+    )
+    product_uom_qty = fields.Float(
+        'Demand', compute='_compute_picking_quantity',
```

Comment:
Please break line for `compute=`
https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#idioms

Reply:
Well, guidelines doesn't specify this strictly, but it's harmless as well. We plan to introduce auto-formatting soon, so this kind of discussions will be outdated :) If you are coming to OCA days, you can join the track.